pkg/queue/manager.go

```diff
@@ -48,17 +53,59 @@ type Manager struct {
 	clusterQueues map[string]ClusterQueue
 	localQueues   map[string]*LocalQueue
 
+	wm        sync.RWMutex
+	queue     workqueue.Interface
+	snapshots map[string][]kueue.ClusterQueuePendingWorkload
```

Comment:
make sure the snapshots don't leak - we need to call `DeleteSnapshot(cq)` when the corresponding cq is deleted.